staker/stakerapp.go

```diff
@@ -669,6 +676,102 @@ func (app *StakerApp) checkTransactionsStatus() error {
 			// we crashed after succesful send to babaylon, restart checking for unbonding signatures
 			app.wg.Add(1)
 			go app.checkForUnbondingTxSignaturesOnBabylon(stakingTxHash)
+		} else if localInfo.stakingTxState == proto.TransactionState_DELEGATION_ACTIVE {
+			// delegation was send to Babylon and activated by covenants, check wheter we:
+			// - did not spent tx beore restart
+			// - did not sent unbonding tx before restart
+			stakingTxHash := localInfo.stakingTxHash
+			tx, _ := app.mustGetTransactionAndStakerAddress(stakingTxHash)
+
+			// 1. First check if staking output is still unspent on BTC chain
+			stakingOutputSpent, err := app.wc.OutputSpent(stakingTxHash, tx.StakingOutputIndex)
+
+			if err != nil {
+				return err
+			}
+
+			if !stakingOutputSpent {
+				continue
+			}
+
+			// 2. Staking output has been spent, we need to check whether this is unbonding
+			// or withdrawal transaction
+			unbondingTxHash := tx.UnbondingTxData.UnbondingTx.TxHash()
+
+			_, unbondingTxStatus, err := app.wc.TxDetails(
+				&unbondingTxHash,
+				// unbonding tx always have only one output
+				tx.UnbondingTxData.UnbondingTx.TxOut[0].PkScript,
+			)
+
+			if err != nil {
+				return err
+			}
+
+			if unbondingTxStatus == walletcontroller.TxNotFound {
+				// no unbonding tx on chain and staking output already spent, most probably
+				// staking transaction has been withdrawn, update state in db
+				if err := app.txTracker.SetTxSpentOnBtc(stakingTxHash); err != nil {
+					return err
+				}
```

Comment:
seems we can have `app.MustSetTxSpentOnBtc` to avoid embedded if condition and panic upon err